common/events/events_io.cpp

```diff
@@ -123,8 +123,18 @@ void AsyncWriterFromWriter::Cancel() {
 	}
 }
 
-ReaderFromAsyncReader::ReaderFromAsyncReader(AsyncReaderFromEventLoopFunc func) :
-	reader_(func(event_loop_)) {
+ReaderFromAsyncReader::ReaderFromAsyncReader() {
+}
+
+mio::ExpectedReaderPtr ReaderFromAsyncReader::Construct(AsyncReaderFromEventLoopFunc func) {
+	shared_ptr<ReaderFromAsyncReader> reader(new ReaderFromAsyncReader);
```

Comment:
Q: `make_shared`?

Reply:
The constructor is private, so I can't use `make_shared` here.